### PR TITLE
boost::optional for LB philox counter

### PR DESCRIPTION
**Pull request by KaiSzuttor**

### CODE REVIEW

File: src/core/utils/Counter.hpp [hunk 1]
```diff
@@ -8,10 +8,7 @@ template <typename T> class Counter {
 private:
   T m_val;
   T m_initial;
-
-private:
   friend class boost::serialization::access;
```
Comment (fweik) [inline src/core/utils/Counter.hpp:11]: If the method is no longer private the fried declaration can be removed too.

Comment (KaiSzuttor) [reply src/core/utils/Counter.hpp:11]: it's actually still private, I just removed a redundant ```private```

**Review by fweik — DISMISSED** [review]

**Comment by fweik** [comment]
@KaiSzuttor please look at this, seems to be a genuine error

**Review by fweik — APPROVED** [review]

**Pull request merged**